### PR TITLE
integration test

### PR DESCRIPTION
**Pull request by nithinraok**
**Change log:**

- Added integration test for speaker recognition
- Added .tar.gz file for an4 data
- Added sample yaml file for testing 

Signed-off-by: Nithin Rao <nithinrao.koluguri@gmail.com>

### CODE REVIEW

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+from ruamel.yaml import YAML
+
+import nemo
+import nemo.collections.asr as nemo_asr
+
+logging = nemo.logging
+
+
+@pytest.mark.usefixtures("neural_factory")
+class TestSpeakerRecognitonPytorch(TestCase):
```
Comment (blisc) [inline tests/integration/test_speakerRecognition_gradient_step_and_eval.py:34]: I believe that tests should no longer inherit from TestCase. @tkornuta-nvidia any comments here?

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        "normalize": "per_feature",
+        "window_size": 0.02,
+        "window_stride": 0.01,
+        "window": "hann",
+        "features": 64,
+        "n_fft": 512,
+        "frame_splicing": 1,
+        "dither": 0.00001,
+        "stft_conv": False,
+    }
```
Comment (blisc) [inline tests/integration/test_speakerRecognition_gradient_step_and_eval.py:46]: What is the point of this if you have it inside your yaml?

File: tests/data/quartznet_spkr_test.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
+model: "GramVoxNet"
```
Comment (blisc) [inline tests/data/quartznet_spkr_test.yaml:1]: Is this file any different from your configs inside of examples?

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+            "normalize": "per_feature",
+            "window_size": 0.02,
+            "window_stride": 0.01,
+            "window": "hann",
+            "features": 64,
+            "n_fft": 512,
+            "frame_splicing": 1,
+            "dither": 0.00001,
+            "stft_conv": False,
+        }
```
Comment (blisc) [inline tests/integration/test_speakerRecognition_gradient_step_and_eval.py:110]: Isn't this in your yaml?

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+            'frame_splicing': 1,
+            'features': 64,
+            'window_size': 0.02,
+            'n_fft': 512,
+            'dither': 1e-05,
+            'window': 'hann',
+            'sample_rate': 16000,
+            'normalize': 'per_feature',
+            'window_stride': 0.01,
+        }
```
Comment (blisc) [inline tests/integration/test_speakerRecognition_gradient_step_and_eval.py:162]: Isn't this in your yaml?

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        )
+        self.nf.reset_trainer()
+
+        # Assert that training loss went down
+        assert loss_list[-1] < loss_list[0]
+
+    @pytest.mark.integration
+    def test_quartznet_speaker_reco_eval(self):
+        """Integration test that tests EvaluatorCallback and NeuralModuleFactory.eval().
+        """
```
Comment (blisc) [inline tests/integration/test_speakerRecognition_gradient_step_and_eval.py:146]: Can you add some asserts to this test? What does this test add to your testing suite that isn't already covered?

Comment (nithinraok) [reply tests/integration/test_speakerRecognition_gradient_step_and_eval.py:146]: this is just to see if eval call back was working. Yes I agree no assertions or checks. Will remove

File: tests/data/quartznet_spkr_test.yaml (hunk 3 above)
Comment (nithinraok) [reply tests/data/quartznet_spkr_test.yaml:1]: Yeah smaller emb size

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py (hunk 2 above)
Comment (nithinraok) [reply tests/integration/test_speakerRecognition_gradient_step_and_eval.py:46]: Noted

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py (hunk 4 above)
Comment (nithinraok) [reply tests/integration/test_speakerRecognition_gradient_step_and_eval.py:110]: Removed eval case

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py (hunk 5 above)
Comment (nithinraok) [reply tests/integration/test_speakerRecognition_gradient_step_and_eval.py:162]: Removed

File: tests/integration/test_speakerRecognition_gradient_step_and_eval.py (hunk 1 above)
Comment (tkornuta-nvidia) [reply tests/integration/test_speakerRecognition_gradient_step_and_eval.py:34]: Yap, we have switched to pytest - so you don't need to have this inheritance.

Still, pytest handles that too. But you won't be able to use some of the more advanced fixtures.

Comment (nithinraok) [reply tests/integration/test_speakerRecognition_gradient_step_and_eval.py]: Noted

**Pull request merged**